src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -387,7 +389,8 @@ public boolean parseArguments(final PrintStream messageStream, final String[] ar
 
         //check if special short circuiting arguments are set
         if (isSpecialFlagSet(parsedArguments, SpecialArgumentsCollection.HELP_FULLNAME)) {
-            messageStream.print(usage(true));
+            // TODO: should a new SpecialArgumentCollection being set to display hidden arguments?
```

Comment:
@magicDGS Yes, I think we want to add special argument called "showHidden", and annotate it as....Advanced. We'd then need to also add support for Advanced args in the usage output, so they show up in a special group (we need to do that anyway, but this will force our hand - if this is more than you signed up for I'll be happy to do it). @vdauwera any input on what the language should be for the "Advanced" argument group should be, maybe "For advanced/experimental use only" or something ?